planet/src/main/webapp/places.js

```diff
@@ -342,36 +336,33 @@ function listResults() {
         div1.appendChild(div2); 
         element.appendChild(div1);
     }
-    // Add search keyword to header
+    // Add search keyword to header 
     document.getElementById('greeting').innerHTML = 'Find a place: ' + document.getElementById('pac-input').value;
 }
 
 /** Toggle save icon on click */
 function savePlace(x) {
+    let placeID = $(x).attr('name');
+    let userID = currentUser.uid; 
```

Comment:
Are you sure that currentUser is not 'null'? (the user is logged in)

Reply:
When the user is logged in it isn't null

Reply:
We will have to implement something that tells the user they have to be logged in to save places

Reply:
If the user is not logged in, you will get an error for trying to access the property uid of undefined. I suggest you add a condition to check that (currentUser !== null).

Reply:
Will do in next PR!